c2rust-analyze/src/context.rs

```diff
@@ -463,15 +463,16 @@ impl<'a, 'tcx> AnalysisCtxt<'a, 'tcx> {
         }
     }
 
-    pub fn project(&self, lty: LTy<'tcx>, proj: &PlaceElem<'tcx>) -> LTy<'tcx> {
+    pub fn projection_lty(&self, lty: LTy<'tcx>, proj: &PlaceElem<'tcx>) -> LTy<'tcx> {
         let projection_lty = |_lty: LTy, adt_def: AdtDef, field: Field| {
             let field_def = &adt_def.non_enum_variant().fields[field.index()];
             let field_def_name = field_def.name;
             eprintln!("projecting into {adt_def:?}.{field_def_name:}");
-            let res = *self.gacx.field_tys.get(&field_def.did).unwrap_or_else(|| {
-                panic!("Could not find {adt_def:?}.{field_def_name:?} in field type map")
-            });
-            res
+            let field_lty: &crate::labeled_ty::LabeledTyS<PointerId> =
```

Comment:
Is this annotation autogenerated?  It would be clearer to use the `LTy` here, or `LabeledTy<_>`, rather than the more verbose and less common `&LabeledTyS<_>`.  Or else remove the annotation and add the `*` back to the RHS.

Reply:
Yes, it was auto-generated. Updated in 4b2537b4